lib/board.rb

```diff
@@ -4,49 +4,54 @@ class Board
   PLAYER_ONE = 1
   PLAYER_TWO = 2
 
-  attr_reader :errors, :positions
+  attr_reader :error, :positions
 
   def initialize(positions = EMPTY_BOARD)
-    @errors = []
+    @error = nil
     @positions = positions
   end
 
   def available_positions
     available = []
-    @positions.each_with_index { |pos, idx| available << -pos * (idx + 1) }
+    @positions.each_with_index { |position, index| available << index if position < 0 }
     available.reject { |pos| pos < 0 }
```

Comment:
available.select(&:positive?)

Reply:
I think that would work, no?

Reply:
Hmm, I didn't even realise ruby had the ability to do function capturing, or whatever that notation is. Something to go and look up.

Sadly, I need to include zero, which `positive?` excludes, but `available.reject(&:negative?)` does the trick. I think it looks cleaner and shows intent better too. Thanks.